pkg/volume/git_repo/git_repo_test.go

```diff
@@ -55,106 +56,408 @@ func TestCanSupport(t *testing.T) {
+						Repository: gitUrl,
+						Directory:  "target_dir",
+					},
+				},
+			},
+			expecteds: []expectedCommand{
+				{
+					cmd: []string{"git", "clone", gitUrl, "target_dir"},
+					dir: "",
+				},
+			},
+			pSpec:             validPluginSpec,
+			isExpectedFailure: false,
+		},
+		"only-git-clone": {
+			vol: &api.Volume{
+				Name: "vol1",
+				VolumeSource: api.VolumeSource{
+					GitRepo: &api.GitRepoVolumeSource{
+						Repository: gitUrl,
+					},
+				},
+			},
+			expecteds: []expectedCommand{
+				{
+					cmd: []string{"git", "clone", gitUrl},
+					dir: "",
+				},
+			},
+			pSpec:             validPluginSpec,
+			isExpectedFailure: false,
+		},
+		"no-target-dir": {
+			vol: &api.Volume{
+				Name: "vol1",
+				VolumeSource: api.VolumeSource{
+					GitRepo: &api.GitRepoVolumeSource{
+						Repository: gitUrl,
+						Revision:   revision,
+						Directory:  "",
+					},
+				},
+			},
+			expecteds: []expectedCommand{
+				{
+					cmd: []string{"git", "clone", gitUrl},
+					dir: "",
+				},
+				{
+					cmd: []string{"git", "checkout", revision},
+					dir: "/kubernetes",
+				},
+				{
+					cmd: []string{"git", "reset", "--hard"},
+					dir: "/kubernetes",
+				},
+			},
+			pSpec:             validPluginSpec,
+			isExpectedFailure: false,
+		},
+		"current-dir": {
+			vol: &api.Volume{
+				Name: "vol1",
+				VolumeSource: api.VolumeSource{
+					GitRepo: &api.GitRepoVolumeSource{
+						Repository: gitUrl,
+						Revision:   revision,
+						Directory:  ".",
+					},
+				},
+			},
+			expecteds: []expectedCommand{
+				{
+					cmd: []string{"git", "clone", gitUrl, "."},
+					dir: "",
+				},
+				{
+					cmd: []string{"git", "checkout", revision},
+					dir: "",
+				},
+				{
+					cmd: []string{"git", "reset", "--hard"},
+					dir: "",
+				},
+			},
+			pSpec:             validPluginSpec,
+			isExpectedFailure: false,
+		},
+		"invalid-dir": {
+			vol: &api.Volume{
+				Name: "vol1",
+				VolumeSource: api.VolumeSource{
+					GitRepo: &api.GitRepoVolumeSource{
+						Repository: gitUrl,
+						Revision:   revision,
+						Directory:  ".",
+					},
+				},
+			},
+			expecteds: []expectedCommand{
+				{
+					cmd: []string{"git", "clone", gitUrl, "."},
+					dir: "",
+				},
+				{
+					cmd: []string{"git", "checkout", revision},
+					dir: "target_dir",
+				},
+				{
+					cmd: []string{"git", "reset", "--hard"},
+					dir: "target_dir",
+				},
+			},
+			pSpec:             validPluginSpec,
+			isExpectedFailure: true,
+		},
+		"invalid-expecteds": {
+			vol: &api.Volume{
+				Name: "vol1",
+				VolumeSource: api.VolumeSource{
+					GitRepo: &api.GitRepoVolumeSource{
+						Repository: gitUrl,
+						Revision:   revision,
+						Directory:  "target_dir",
+					},
+				},
+			},
+			expecteds: []expectedCommand{
+				{
+					cmd: []string{"git", "clone", gitUrl},
+					dir: "",
+				},
+			},
+			pSpec:             validPluginSpec,
+			isExpectedFailure: true,
+		},
+		"invalid-plugin-name": {
```

Comment:
invalid plugin name isn't really a scenario in scope for this test -- that's something the tests for plugin loading cover.